server/layer.go

```diff
@@ -96,7 +96,7 @@ func (l *Layer) Remove() error {
 	}
 
 	for _, m := range ms {
-		for _, layer := range append(m.Layers, m.Config) {
+		for _, layer := range append(m.Layers, &m.Config) {
```

Comment:
This is dangerous. If m.Config is invalid, i.e. `Digest == ""`, this will continue and eventually call `os.Remove("~/.ollama/models/blobs")`